system/Database/OCI8/Connection.php

```diff
@@ -0,0 +1,720 @@
+
+        $retVal = [];
+
+        foreach ($query as $row) {
+            $obj                      = new stdClass();
+            $obj->constraint_name     = $row->CONSTRAINT_NAME;
+            $obj->table_name          = $row->TABLE_NAME;
+            $obj->column_name         = $row->COLUMN_NAME;
+            $obj->foreign_table_name  = $row->FOREIGN_TABLE_NAME;
+            $obj->foreign_column_name = $row->FOREIGN_COLUMN_NAME;
+            $retVal[]                 = $obj;
+        }
+
+        return $retVal;
+    }
+
+    /**
+     * Returns platform-specific SQL to disable foreign key checks.
+     *
+     * @return string
+     */
+    protected function _disableForeignKeyChecks()
+    {
+        return <<<'SQL'
+            BEGIN
+              FOR c IN
+              (SELECT c.owner, c.table_name, c.constraint_name
+               FROM user_constraints c, user_tables t
+               WHERE c.table_name = t.table_name
+               AND c.status = 'ENABLED'
+               AND c.constraint_type = 'R'
+               AND t.iot_type IS NULL
+               ORDER BY c.constraint_type DESC)
+              LOOP
+                dbms_utility.exec_ddl_statement('alter table "' || c.owner || '"."' || c.table_name || '" disable constraint "' || c.constraint_name || '"');
+              END LOOP;
+            END;
+            SQL;
+    }
+
+    /**
+     * Returns platform-specific SQL to enable foreign key checks.
+     *
+     * @return string
+     */
+    protected function _enableForeignKeyChecks()
+    {
+        return <<<'SQL'
+            BEGIN
+              FOR c IN
+              (SELECT c.owner, c.table_name, c.constraint_name
+               FROM user_constraints c, user_tables t
+               WHERE c.table_name = t.table_name
+               AND c.status = 'DISABLED'
+               AND c.constraint_type = 'R'
+               AND t.iot_type IS NULL
+               ORDER BY c.constraint_type DESC)
+              LOOP
+                dbms_utility.exec_ddl_statement('alter table "' || c.owner || '"."' || c.table_name || '" enable constraint "' || c.constraint_name || '"');
+              END LOOP;
+            END;
+            SQL;
+    }
+
+    /**
+     * Get cursor. Returns a cursor from the database
+     *
+     * @return resource
+     */
+    public function getCursor()
+    {
+        return $this->cursorId = oci_new_cursor($this->connID);
+    }
+
+    /**
+     * Executes a stored procedure
+     *
+     * @param string $procedureName procedure name to execute
+     *
+     * @return mixed
+     *
+     * params array keys
+     *
+     * KEY      OPTIONAL  NOTES
+     * name     no        the name of the parameter should be in :<param_name> format
+     * value    no        the value of the parameter.  If this is an OUT or IN OUT parameter,
+     *                    this should be a reference to a variable
+     * type     yes       the type of the parameter
+     * length   yes       the max size of the parameter
```

Comment:
Since this description is about parameters, please modify it to the `@param` format.

Reply:
Thanks, the feedback.

Fixed it!.